packages/js-dpp/lib/dataTrigger/rewardShareDataTriggers/createMasternodeRewardSharesDataTrigger.js

```diff
@@ -0,0 +1,84 @@
+const DataTriggerConditionError = require('../../errors/consensus/state/dataContract/dataTrigger/DataTriggerConditionError');
+const DataTriggerExecutionResult = require('../DataTriggerExecutionResult');
+
+const MAX_PERCENTAGE = 10000;
+
+/**
+ * @param {DocumentCreateTransition} documentTransition
+ * @param {DataTriggerExecutionContext} context
+ * @return {Promise<DataTriggerExecutionResult>}
+ */
+async function createMasternodeRewardSharesDataTrigger(
+  documentTransition,
+  context,
+) {
+  const {
+    payToId,
+    percentage,
+  } = documentTransition.getData();
+
+  const ownerId = context.getOwnerId();
+
+  const result = new DataTriggerExecutionResult();
+
+  const identity = await context.getStateRepository().fetchIdentity(payToId);
+  if (identity !== null) {
+    const error = new DataTriggerConditionError(
+      context.getDataContract().getId().toBuffer(),
+      documentTransition.getId().toBuffer(),
+      `Identity ${payToId.toString()} already exists`,
+    );
+
+    error.setOwnerId(ownerId);
+    error.setDocumentTransition(documentTransition);
+
+    result.addError(error);
+  }
+
+  const documents = await context.getStateRepository().fetchDocuments(
+    context.getDataContract().getId(),
+    documentTransition.getType(),
+    {
+      where: [
+        ['$ownerId', '==', ownerId],
+      ],
+    },
+  );
+
+  const totalPercent = documents
+    .reduce((prevValue, document) => prevValue + document.data.percentage, percentage);
+
+  if (totalPercent > MAX_PERCENTAGE) {
+    const error = new DataTriggerConditionError(
+      context.getDataContract().getId().toBuffer(),
+      documentTransition.getId().toBuffer(),
+      `Percentage can not be more than ${MAX_PERCENTAGE}`,
+    );
+
+    error.setOwnerId(ownerId);
+    error.setDocumentTransition(documentTransition);
+
+    result.addError(error);
+  }
+
+  const smlStore = await context.getStateRepository().fetchSMLStore();
```

Comment:
I would check it first. If this is invalid then all other checks just wasting resources.